doc/user_guide/model.rst

```diff
@@ -775,43 +791,80 @@ gaussian noise and proceed to fit as in the previous example.
     >>> line.coefficients.std
     (0.0081710549764721901, 1.4117294994070277)
 
-Because the noise is heterocedastic, the least squares optimizer estimation is
-biased. A more accurate result can be obtained by using weighted least squares
-instead that, although still biased for poissonian noise, is a good
-approximation in most cases.
+Because the noise is heteroscedastic, the least squares optimizer estimation is
+biased. A more accurate result can be obtained with weighted least squares,
+where the weights are proportional to the inverse of the noise variance.
+Although this is still biased for Poisson noise, it is a good approximation
+in most cases.
 
 .. code-block:: python
 
    >>> s.estimate_poissonian_noise_variance(
-   ...     expected_value=hs.signals.Signal1D(np.arange(300)))
-   >>> m.fit()
+   ...     expected_value=hs.signals.Signal1D(np.arange(300))
+   ... )
+   >>> m.fit(weights="inverse_variance")
    >>> line.coefficients.value
    (1.0004224896604759, -0.46982916592391377)
    >>> line.coefficients.std
    (0.0055752036447948173, 0.46950832982673557)
 
+Instead of using the inverse variance, you can also provide your own
+weights as a ``np.ndarray``:
+
+.. code-block:: python
+
+   >>> weights = np.ones_like(s.data)
+   >>> m.fit(weights=weights)
+
+.. warning::
+
+    When the attribute ``metadata.Signal.Noise_properties.variance``
+    is defined, the default behaviour is to perform a weighted least-squares
+    fit using the inverse variance.
+
+    In this scenario, to explicitly disable weighting, pass ``m.fit(weights=None)``.
```

Comment:
> Why would one want to disable weighting if the variance is defined? Also, wouldn't that be equivalent to remove variance from metadata?

Yes - but for example, say you estimated the Poisson noise, and wanted to compare the results of weighted vs. unweighted fits.

I would say it is easier to just run `m.fit()` and `m.fit(weights=None)` to compare the results, rather than estimating the variance then unsetting it and (potentially) having to set it again.

Reply:
I see, but the question is: why would you want to compare one with the other when one way is right and the other is not? Some may want to do for, say, educational purposes, but I don't think that it is worth adding a feature for that since that minority can always simply delete the variance metadata.

Reply:
Sure. I'll tidy it up

Reply:
Just to be clear here - in removing the `weights` kwarg @francisco-dlp, are you happy for #1627 to just be closed as it is? Perhaps with some clearer documentation that once the variance is set, all fits are weighted until it is unset?

Reply:
I agree that it is a good idea to close #1627 by just documenting how to do it.

Reply:
Great thanks - I will do that.